sampling.js

```diff
@@ -1,149 +1,192 @@
+// Bin Size and num_samples could be dynamic
+// Either derived from file data (size, genome type, etc) or defined by user input (config file, env values)
 const TARGET_BIN_SIZE = 10000;
 const SECONDARY_BIN_SIZE = 5000;
 const TERTIARY_BIN_SIZE = 2500;
 const NUM_SAMPLES = 20;
 
+// Confirm if inRegions can be narrowed to a specific region
+// Otherwise regions code is unchanged
 function sample(inRegions) {
-
-  let idealRegions = [];
-  const secondaryRegions = [];
-  const tertiaryRegions = [];
-
-  for (const region of inRegions) {
-    const length = region.end - region.start;
-
-    if (length >= TARGET_BIN_SIZE) {
-      idealRegions.push(region);
-    }
-    else if (length >= SECONDARY_BIN_SIZE) {
-      secondaryRegions.push(region);
-    }
-    else if (length >= TERTIARY_BIN_SIZE) {
-      tertiaryRegions.push(region);
-    }
-  }
-
-  if (idealRegions.length < NUM_SAMPLES) {
-    const expanded = expandRegions(idealRegions);
-
-    if (expanded.length > idealRegions.length) {
-      idealRegions = expanded;
-    }
-  }
-
-  let sampledRegions = sampleFromRegions(idealRegions, NUM_SAMPLES, TARGET_BIN_SIZE);
-
-  if (sampledRegions.length < NUM_SAMPLES) {
-    const remaining = NUM_SAMPLES - sampledRegions.length;
-    // readRatio increases the number of regions so we still get the desired number of reads sampled
-    const readRatio = Math.floor(TARGET_BIN_SIZE/SECONDARY_BIN_SIZE);
-    const batch = sampleFromRegions(secondaryRegions, remaining*readRatio, SECONDARY_BIN_SIZE);
-    sampledRegions = [...sampledRegions, ...batch];
-  }
-
-  if (sampledRegions.length < NUM_SAMPLES) {
-    const remaining = NUM_SAMPLES - sampledRegions.length;
-    const readRatio = Math.floor(TARGET_BIN_SIZE/TERTIARY_BIN_SIZE);
-    const batch = sampleFromRegions(tertiaryRegions, remaining*readRatio, TERTIARY_BIN_SIZE);
-    sampledRegions = [...sampledRegions, ...batch];
-  }
-
-  return sampledRegions.sort(function (a, b) {
-    if (a.name == b.name) {
-      return ((a.start < b.start) ? -1 : ((a.start > b.start) ? 1 : 0));
-    }
-    else {
-      return ((a.name < b.name) ? -1 : ((a.name > b.name) ? 1 : 0));
-    }
-  });
+	let idealRegions = [];
```

Comment:
Lines 11-130 are only formatting updates, and replacing default sampling function with static sampling function